modules/types-registry/types-registry-sdk/src/api.rs

```diff
@@ -0,0 +1,80 @@
+//! `TypesRegistryApi` trait definition.
+//!
+//! This trait defines the public API for the `types-registry` module.
+//! All methods require a `SecurityCtx` for authorization and access control.
+
+use async_trait::async_trait;
+use modkit_security::SecurityCtx;
+
+use crate::error::TypesRegistryError;
+use crate::models::{GtsEntity, ListQuery};
+
+/// Public API trait for the `types-registry` module.
+///
+/// This trait can be consumed by other modules via `ClientHub`:
+/// ```ignore
+/// let client = hub.get::<dyn TypesRegistryApi>()?;
+/// let entity = client.get(&ctx, "gts.acme.core.events.user_created.v1~").await?;
+/// ```
+///
+/// All methods require a `SecurityCtx` for proper authorization and access control.
+#[async_trait]
+pub trait TypesRegistryApi: Send + Sync {
```

Comment:
PROPOSAL: Starting with this PR, I suggest renaming all these clients from XXXXApi to XXXClient, as the Api suffix is confusing. @Artifizer what do you think?